dns.go

```diff
@@ -45,10 +46,34 @@ type DNSListResponse struct {
 	ResultInfo `json:"result_info"`
 }
 
+// nontransitionalLookup implements the nontransitional processing as specified in
+// Unicode Technical Standard 46 with almost all checkings off to maximize user freedom.
+var nontransitionalLookup = idna.New(
+	idna.MapForLookup(),
+	idna.StrictDomainName(false),
+	idna.ValidateLabels(false),
+)
+
+// normalizeDomainName tries to convert IDNs (international domain names)
+// from Unicode form to Punycode, using non-transitional process specified
+// in UTS 46.
+//
+// Confusinigly, this is in the opposite direction of normalizeZoneName
+// to work around the current quirks of the Cloudflare servers.
```

Comment:
this line isn't quite accurate; it's nothing to do with _servers_ but the Cloudflare DNS API service itself. if we're going to keep this line (which my vote is to remove this and instead use a better type/method structure) we need to reflect reality.

Reply:
i.e. implement the [stringer interface](https://pkg.go.dev/fmt#Stringer) for these types

Reply:
I will update or remove the comments accordingly. Currently, "server" is being used in `cloudflare.go` (and maybe other files) to refer to "API service" but I will leave the updates to another PR.